Source/Timeline/TimelinePagerView.swift

```diff
@@ -20,7 +20,17 @@ public final class TimelinePagerView: UIView, UIGestureRecognizerDelegate, UIScr
   public weak var dataSource: EventDataSource?
   public weak var delegate: TimelinePagerViewDelegate?
 
-  public private(set) var calendar: Calendar = Calendar.autoupdatingCurrent
+  public var calendar: Calendar = Calendar.autoupdatingCurrent {
+    didSet {
+      // changing timezone in loaded pages
+      pagingViewController.children.forEach {
+        let vc = $0 as! TimelineContainerController
```

Comment:
My main suggestion for this pull request would be instead of modifying the existing `TimelineContainerController `s , just create a set of new ones with the new calendar and reload with the new data.

It would be much simpler programmatically, as the newly created controllers would have a clean state, instead of a modified one, which would make it easier to reason about...

... And would remove the need to use hacks such as in `updateTimeline` and `func pageViewController(_ pageViewController: UIPageViewController, didFinishAnimating finished: Bool, previousViewControllers: [UIViewController], transitionCompleted completed: Bool)`